### ISSUE TITLE
TimeDistributed(InceptionV3)-LSTM Evalutaion Requires Same Batch Size as Training to Produce Correct Results

### ISSUE THREAD

**Issue by TheRealYoussef** [keras-team/keras#8364]
-Keras 2.0.9, TensorFlow backend-

I am building a model which classifies videos using a TimeDistributed(InceptionV3)-LSTM architecture:

```
K.set_learning_phase(1)
video = Input(shape=(10, 299, 299, 3), name='video_input')
cnn = InceptionV3(
    weights='imagenet',
    include_top=False,
    pooling='max'
)
encoded_frame = TimeDistributed(cnn)(video)
encoded_vid = LSTM(1024)(encoded_frame)
outputs = Dense(16, activation='softmax')(encoded_vid)
model = Model(inputs=[video],outputs=outputs)
model.compile(loss='categorical_crossentropy', optimizer='sgd', metrics=['accuracy'])
```
The first problem is that to train this model without errors, setting the learning phase to 1 is required. The second problem is that if I train the model using batch size BATCH:

```
BATCH = 8
model.fit(x=x_train, 
               y=y_train,
               batch_size=BATCH,
               epochs=10,
               validation_data=(x_val, y_val))
```
Then evaluating or predicting requires the same batch size, or otherwise it produces wrong results.

`model.evaluate(x_val, y_val, BATCH)`
Produces correct results.

`model.evaluate(x_val, y_val, 1)`
Produces wrong results.

https://gist.github.com/TheRealYoussef/4dd8ad822389be2179cc1290a4c1b9d6

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by TheRealYoussef** [comment]
> building an identical model but changing the input shape to be a batch size of one

How do I change the input shape to be a batch size of one? When defining the input shape, the batch size isn't included. Batch size is only introduced when calling `model.fit()`

**Comment by TheRealYoussef** [comment]
Update: Giving up on Keras because of this issue, I went over to TensorFlow directly to try out a similar model. Basically it is formed of a reshape layer to change from shape (batch_size, num_frames, width, height, channels) to shape (batch_size * num_frames, width, height, channels). Following this reshape layer is the Inception V3 model, followed by another reshape layer to change the output of the Inception V3 to (batch_size, num_frames, 2048), the 2048 being the number of features output by the Inception V3. Following this is an LSTM layer, followed finally by the dense layer for the final classification. This is basically the same thing as the Keras model above. Anyways, the same thing ended up happening (evaluation required the same batch size as training). This turned out to be because of the batch normalization layers and the different ways it acts during training and testing. During training, batch normalization uses the statistics of the batch, while during testing, it uses the moving average and standard deviation, which it is supposed to learn during training, but wasn't. Finally, I found out that I simply need to add code which explicitly tells TensorFlow to learn these weights:
```
update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
with tf.control_dependencies(update_ops):
    train_op = tf.train.GradientDescentOptimizer(0.01).minimize(cross_entropy)
```
